Copyright (c) 2014 Citrix Systems, Inc.

diff -uNr mceusb.orig/mceusb.c mceusb/mceusb.c
--- mceusb.orig/mceusb.c	2014-01-31 21:27:20.000000000 +0000
+++ mceusb/mceusb.c	2014-01-31 21:27:29.000000000 +0000
@@ -284,7 +284,8 @@
 	/* Philips Infrared Transceiver - Spinel plus */
 	{ USB_DEVICE(VENDOR_PHILIPS, 0x0613) },
 	/* Philips eHome Infrared Transceiver */
-	{ USB_DEVICE(VENDOR_PHILIPS, 0x0815) },
+	{ USB_DEVICE(VENDOR_PHILIPS, 0x0815),
+          .driver_info = MCE_GEN2_TX_INV },
 	/* Philips/Spinel plus IR transceiver for ASUS */
 	{ USB_DEVICE(VENDOR_PHILIPS, 0x206c) },
 	/* Philips/Spinel plus IR transceiver for ASUS */
@@ -317,7 +318,8 @@
 	/* Gateway eHome Infrared Transceiver */
 	{ USB_DEVICE(VENDOR_GATEWAY, 0x3009) },
 	/* Mitsumi */
-	{ USB_DEVICE(VENDOR_MITSUMI, 0x2501) },
+	{ USB_DEVICE(VENDOR_MITSUMI, 0x2501),
+	  .driver_info = MCE_GEN2_TX_INV },
 	/* Topseed eHome Infrared Transceiver */
 	{ USB_DEVICE(VENDOR_TOPSEED, 0x0001),
 	  .driver_info = MCE_GEN2_TX_INV },
